src/lib/components/workspace/member-info/MemberInfo.svelte

```diff
@@ -0,0 +1,188 @@
+<script lang="ts">
+  import MemberDropdown from "$lib/components/dropdown/MemberDropdown.svelte";
+  import MemberWorkspace from "../member-worspace/MemberWorkspace.svelte";
+  import { TeamRole, WorkspaceRole } from "$lib/utils/enums/team.enum";
+  import type {
+    TeamServiceMethods,
+    userDetails,
+    workspaceDocumentWithPosition,
+  } from "$lib/utils/interfaces";
+  import type { MemberPopType } from "$lib/utils/types/common.type";
+
+  export let user: userDetails;
+  export let teamRole: TeamRole;
+  export let workspaces: workspaceDocumentWithPosition[];
+  export let hasPermission: boolean = false;
+  export let userType: TeamRole;
+  export let handlePopup: (flag: boolean, popType: MemberPopType) => void;
+  export let owner: boolean = false;
+  export let teamServiceMethods: TeamServiceMethods = {};
+  export let userId: string;
+  export let handleMemberPopUpSuccess = (flag: boolean) => {};
+  export let getPermissionsData: () => Array<{
+    name: string;
+    id: string;
+    color: string;
+  }>;
+  export let isWorkspaceMemberInfo = false;
+  export let handleDropDownWorkspaceLevel = (
+    currentRole: WorkspaceRole | "remove",
+  ) => {};
+  const handleDropdown = (id: "remove" | TeamRole) => {
+    if (id === "remove") {
+      handlePopup(true, "isMemberRemovePopup");
+    } else if (
+      user.role === TeamRole.TEAM_ADMIN &&
+      id === TeamRole.TEAM_MEMBER
+    ) {
+      handlePopup(true, "isMemberDemotePopup");
+    } else if (
+      user.role === TeamRole.TEAM_MEMBER &&
+      id === TeamRole.TEAM_ADMIN
+    ) {
+      handlePopup(true, "isMemberPromotePopup");
+    } else if (
+      user.role === TeamRole.TEAM_ADMIN &&
+      id === TeamRole.TEAM_OWNER
+    ) {
+      handlePopup(true, "isMemberOwnershipPopup");
+    }
+  };
+
+  let workspaceCount: number;
+  $: {
+    workspaceCount = 0;
+    if (workspaces) {
+      workspaces.forEach((element) => {
+        if (element.position) {
+          workspaceCount++;
+        }
+      });
+    }
+  }
+</script>
+
+<div>
+  <div class="d-flex tile">
+    <div class="info d-flex">
+      <div class="icon d-flex align-items-center justify-content-center">
+        <span>{user.name[0].toUpperCase()}</span>
+      </div>
+      <div class="name px-2">
+        <span style="font-size:12px;" class="text-whiteColor"
+          >{user.name} {owner ? "(You)" : ""}</span
+        ><br />
+        <span style="font-size:12px;" class="text-textColor">{user.email}</span>
+      </div>
+    </div>
+    <div class="position">
+      {#if (userType === TeamRole.TEAM_OWNER && (isWorkspaceMemberInfo ? teamRole === TeamRole.TEAM_MEMBER : user.role === TeamRole.TEAM_MEMBER)) || (userType === TeamRole.TEAM_ADMIN && isWorkspaceMemberInfo ? teamRole === TeamRole.TEAM_MEMBER : user.role === TeamRole.TEAM_MEMBER)}
+        <MemberDropdown
+          workspaceId={user.workspaceId}
+          id={user.id + "owner-team-acess"}
```

Comment:
Take the access strings from an enum

owner-team-acess
admin-team-access
admin-workspace-member-access

Reply:
resolved